includes/class-micropub-authorize.php

```diff
@@ -35,8 +38,49 @@ public static function init() {
 		add_action( 'send_headers', array( $cls, 'http_header' ) );
 		add_filter( 'host_meta', array( $cls, 'jrd_links' ) );
 		add_filter( 'webfinger_user_data', array( $cls, 'jrd_links' ) );
+		add_filter( 'rest_index', array( $cls, 'rest_index' ) );
+
 		// The WordPress IndieAuth plugin uses priority 30
 		add_filter( 'determine_current_user', array( $cls, 'determine_current_user' ), 31 );
+		add_filter( 'rest_authentication_errors', array( $cls, 'rest_authentication_errors' ), 10 );
+		add_filter( 'rest_post_dispatch', array( $cls, 'return_oauth_error' ), 10, 3 );
+
+	}
+
+	public static function return_oauth_error( $result, $server, $request ) {
```

Comment:
this should probably be named something else, right?

Reply:
micropub_error?